lib/registration.pm

```diff
@@ -150,7 +150,12 @@ sub add_suseconnect_product {
     $version //= '${VERSION_ID}';
     $arch    //= '${CPU}';
     $params  //= '';
-    assert_script_run("SUSEConnect -p $name/$version/$arch $params", $timeout);
+    $timeout //= 300;
+    my $cmd = "SUSEConnect -p $name/$version/$arch $params";
+    if (($name =~ /PackageHub/ && is_sle '>=15') && script_run($cmd, $timeout)) {
+        record_soft_failure 'bsc#1124318 - Fail to get PackageHub Pool Metadata - running the command again as a workaround';
+    }
+    assert_script_run($cmd, $timeout);
```

Comment:
I just realized this is not inside the if. The command will always be executed two times.

Reply:
True. I must not be within the if because the if is only entered if it's about package hub though. Changed again, please take a look at the latest state.